src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php

```diff
@@ -736,6 +736,74 @@ private function getRedis(): \Redis|Relay|\RedisCluster
         return $this->redis;
     }
 
+    public function findAll(?int $limit = null): array
+    {
+        $redis = $this->getRedis();
+        $messages = [];
+
+        try {
+            // Use XRANGE to get all messages from the stream
+            // For Redis compatibility, always pass 4 parameters
+            // When limit is null, use -1 to indicate "all messages"
+            $count = $limit ?? -1;
+            $range = $redis->xRange($this->stream, '-', '+', $count);
+        } catch (\RedisException|\Relay\Exception $e) {
+            throw new TransportException($e->getMessage(), 0, $e);
+        }
+
+        if (false === $range) {
+            if ($error = $redis->getLastError() ?: null) {
+                $redis->clearLastError();
+            }
+            throw new TransportException($error ?? 'Could not read messages from the redis stream.');
+        }
+
+        foreach ($range as $id => $data) {
+            $messages[] = [
+                'id' => $id,
+                'data' => $data,
+            ];
+        }
+
+        return $messages;
+    }
+
+    public function find(mixed $id): ?array
+    {
+        if ($this->autoSetup) {
+            $this->setup();
+        }
+
+        if (!\is_string($id)) {
+            return null;
+        }
+
+        $redis = $this->getRedis();
+
+        try {
+            // Use XRANGE to get a specific message by ID
+            $range = $redis->xRange($this->stream, $id, $id, 1);
+        } catch (\RedisException|\Relay\Exception $e) {
+            throw new TransportException($e->getMessage(), 0, $e);
+        }
+
+        if (false === $range || empty($range)) {
```

Comment:
```suggestion
        if (!$range) {
```

We try to avoid using `empty` in the Symfony codebase.